src/mono/mono/metadata/marshal-ilgen.c

```diff
@@ -195,7 +195,7 @@ mono_mb_emit_save_args (MonoMethodBuilder *mb, MonoMethodSignature *sig, gboolea
 		mono_mb_emit_byte (mb, CEE_STIND_I);
 		/* tmp = tmp + sizeof (gpointer) */
 		if (sig->param_count)
-			mono_mb_emit_add_to_local (mb, tmp_var, TARGET_SIZEOF_VOID_P);
+			mono_mb_emit_add_to_local (mb, GINT_TO_UINT16 (tmp_var), TARGET_SIZEOF_VOID_P);
```

Comment:
Maybe we can hold off on changing `marshal*.[ch]` until #69208 goes in? /cc @naricc